pyiceberg/table/__init__.py

```diff
@@ -797,6 +850,9 @@ def location(self) -> str:
     def last_sequence_number(self) -> int:
         return self.metadata.last_sequence_number
 
+    def next_sequence_number(self) -> int:
+        return self.last_sequence_number + 1 if self.metadata.format_version > 1 else INITIAL_SEQUENCE_NUMBER
```

Comment:
Why have both `next_sequence_number` and `_next_sequence_number` just below this? They also have slightly different logic.